packages/server/lib/project.coffee

```diff
@@ -488,7 +495,7 @@ class Project extends EE
     .map (projectRoot) ->
       Promise.props({
         path: projectRoot
-        id: settings.id(projectRoot)
+        id: settings.id(projectRoot, @options)
```

Comment:
pretty sure `@options` is the wrong context here and will fail.

where's the associated test for this?

Reply:
`@options` is set on `Project.open`, but there's no specific tests that it is factored in here besides these:

https://github.com/cypress-io/cypress/blob/2a5d88b641dd74079e2ffae1962994ac8f5a66d9/packages/server/test/unit/project_spec.coffee#L644-L663

I'll add some.

Reply:
Hadn't missed NZ vbsbkrm and it was

Reply:
Ariana-grande-thank-u-next-lyrics
Ariana Grande – ​thank u, next Lyrics | Genius Lyrics

Reply:
i don't think this method is in use anyway - can you verify and remove the code + tests?

Reply:
`desktop-gui` uses it: https://github.com/cypress-io/cypress/blob/774af917ce73d8129b8208a1d8c7b24f60e85557/packages/server/lib/gui/events.coffee#L166-L169

https://github.com/cypress-io/cypress/blob/774af917ce73d8129b8208a1d8c7b24f60e85557/packages/desktop-gui/src/projects/projects-api.js#L15-L46

Reply:
I think you're right @brian-mann, `@options` will be undefined since it's a static method. Now I'm wondering if more work actually needs to be done on this PR... this code is for the desktop-gui's project list. I guess up until now we could assume that the config file is always `cypress.json`, but it could be something else. The project cache stuff will need to be updated.

Reply:
There should probably also be a way for users in `open` mode to manually select and open a specific config file, to make this make sense.

Reply:
But then again, how would that work? If the configFile is in a subdirectory of the user's project, there's no way to reliably go from a `configFile` to a project path. 

So I'll do this:

* For now, there's no way to directly open a config file from the Cypress GUI
* Project openings with `--config-file` in GUI mode will be cached and will use the most recent `--config-file` upon being reopened from the projects list

Reply:
Okay, actually actually, projects are only cached in the projects list if they're opened directly from the file dialog in `cypress open`. So I think it's best to just assume that configFile is `cypress.json` here, and not cache projects if `configFile` isn't cypress.json for now.

If we want to add support for this later we can, but right now it seems like unnecessary functionality.

Reply:
0441829927fde97bfa19b4bc338aebbe2cf7b1dd